packages/vitest/src/node/config.ts

```diff
@@ -151,6 +136,46 @@ export function resolveConfig(
   if (!resolved.deps.moduleDirectories.includes('/node_modules/'))
     resolved.deps.moduleDirectories.push('/node_modules/')
 
+  resolved.server ??= {}
+  resolved.server.deps ??= {}
+
+  const deprecatedDepsOptions = ['inline', 'external', 'fallbackCJS'] as const
+  deprecatedDepsOptions.forEach((option) => {
+    if (resolved.deps[option] !== undefined) {
+      if (option === 'fallbackCJS') {
+        console.warn(c.yellow(`${c.inverse(c.yellow(' Vitest '))} "deps.${option}" is deprecated. Use "server.deps.${option}" instead`))
+      }
+      else {
+        console.warn(
+          c.yellow(
+         `${c.inverse(c.yellow(' Vitest '))} "deps.${option}" is deprecated. If you rely on vite-node directly, use "server.deps.${option}" instead. Otherwise, consider using "deps.optimizer.web.${option === 'external' ? 'exclude' : 'include'}"`,
+          ),
+        )
+      }
+
+      if (resolved.server.deps![option] === undefined)
+        resolved.server.deps![option] = resolved.deps[option] as any
+    }
+  })
+
+  // vitenode will try to import such file with native node,
+  // but then our mocker will not work properly
+  if (resolved.server.deps.inline !== true) {
+    // eslint-disable-next-line @typescript-eslint/prefer-ts-expect-error
+    // @ts-ignore ssr is not typed in Vite 2, but defined in Vite 3, so we can't use expect-error
+    const ssrOptions = viteConfig.ssr
+    if (ssrOptions?.noExternal === true && resolved.server.deps.inline == null) {
+      resolved.server.deps.inline = true
+    }
+    else {
+      resolved.server.deps.inline ??= []
+      resolved.server.deps.inline.push(...extraInlineDeps)
+    }
+  }
+
+  resolved.server.deps.moduleDirectories ??= []
+  resolved.server.deps.moduleDirectories.push(...resolved.deps.moduleDirectories)
```

Comment:
I don't see if you merge all compatible options from `deps` option? Like `interopDefault` should also change `server.deps.interopDefault`

Reply:
Looks like interopDefault is part of runtime, not a server, so this is fine